local_publisher.py

```diff
@@ -22,25 +22,39 @@ class LocalPublisher(TerminatorBlock):
 
     def __init__(self):
         super().__init__()
-        self._publisher = None
+        self._publishers = {}
 
     def configure(self, context):
         super().configure(context)
-        self._publisher = NioPublisher(
-            topic="{}.{}".format(self.local_identifier(), self.topic()))
-        self._publisher.open()
+        try:
+            publisher = NioPublisher(
+                topic="{}.{}".format(self.local_identifier(), self.topic()))
+            publisher.open()
+            self._publishers[topic] = publisher
+        except:
+            # topic is dynamic based on signals
+            pass
 
     def stop(self):
         """ Stop the block by closing the underlying publisher """
-        self._publisher.close()
+        for publisher in self._publishers.values():
+            publisher.close()
         super().stop()
 
     def process_signals(self, signals):
         """ Publish each list of signals """
+        topic = self.topic(signals[0])
+        if topic not in self._publishers:
+            # TODO replace this with a defaultdict that automatically
```

Comment:
Lets either get this implemented or at least remove this comment. I didn't intend for this TODO to be left and eventually merged into master.

Reply:
@hansmosh I made a `_create_publisher` method as we discussed, please review.